extensions/standard-processors/tests/unit/SegmentContentTests.cpp

```diff
@@ -0,0 +1,210 @@
+    CHECK(ranges::equal(segment_bytes, calcExpectedSegment(input_data, segment_i, segment_size)));
+    segment_size_sum += segment_bytes.size();
+  }
+  CHECK(original_size == segment_size_sum);
+}
+
+TEST_CASE("SimpleTest", "[NiFi]") {
+  const auto segment_content = std::make_shared<SegmentContent>("SegmentContent");
+  minifi::test::SingleProcessorTestController controller{segment_content};
+
+  segment_content->setProperty(SegmentContent::SegmentSize, "4 B");
+
+  const auto input_data = createByteVector(1, 2, 3, 4, 5, 6, 7, 8, 9);
+  std::string_view input(reinterpret_cast<const char*>(input_data.data()), input_data.size());
+
+  auto trigger_results = controller.trigger(input);
+
+  auto original = trigger_results.at(processors::SegmentContent::Original);
+  auto segments = trigger_results.at(processors::SegmentContent::Segments);
+
+  REQUIRE(segments.size() == 3);
+  REQUIRE(original.size() == 1);
+
+  auto expected_segment_1 = createByteVector(1, 2, 3, 4);
+  auto expected_segment_2 = createByteVector(5, 6, 7, 8);
+  auto expected_segment_3 = createByteVector(9);
+
+  CHECK(controller.plan->getContentAsBytes(*original[0]) == input_data);
+  CHECK(controller.plan->getContentAsBytes(*segments[0]) == expected_segment_1);
+  CHECK(controller.plan->getContentAsBytes(*segments[1]) == expected_segment_2);
+  CHECK(controller.plan->getContentAsBytes(*segments[2]) == expected_segment_3);
+}
+
+TEST_CASE("TransferSmall", "[NiFi]") {
+  const auto segment_content = std::make_shared<SegmentContent>("SegmentContent");
+  minifi::test::SingleProcessorTestController controller{segment_content};
+
+  segment_content->setProperty(SegmentContent::SegmentSize, "4 KB");
+
+  const auto input_data = createByteVector(1, 2, 3, 4, 5, 6, 7, 8, 9);
+  std::string_view input(reinterpret_cast<const char*>(input_data.data()), input_data.size());
+
+  auto trigger_results = controller.trigger(input);
+
+  auto original = trigger_results.at(processors::SegmentContent::Original);
+  auto segments = trigger_results.at(processors::SegmentContent::Segments);
+
+  REQUIRE(segments.size() == 1);
+  REQUIRE(original.size() == 1);
+
+  CHECK(controller.plan->getContentAsBytes(*segments[0]) == input_data);
+  CHECK(controller.plan->getContentAsBytes(*original[0]) == input_data);
+}
+
+TEST_CASE("ExpressionLanguageSupport", "[NiFi]") {
+  const auto segment_content = std::make_shared<SegmentContent>("SegmentContent");
+  minifi::test::SingleProcessorTestController controller{segment_content};
+
+  segment_content->setProperty(SegmentContent::SegmentSize, "${segmentSize}");
+
+  const auto input_data = createByteVector(1, 2, 3, 4, 5, 6, 7, 8, 9);
+  std::string_view input(reinterpret_cast<const char*>(input_data.data()), input_data.size());
+
+  std::unordered_map<std::string, std::string> attributes = {};
+  attributes["segmentSize"] = "4 B";
+  auto trigger_results = controller.trigger(input, std::move(attributes));
```

Comment:
why not
```suggestion
  auto trigger_results = controller.trigger(input, {{"segmentSize", "4 B"}});
```
?

Reply:
good idea [review changes#diff-b8f50109b2637cc95d2f47b946e4b8f961d488e508046f2922f175c5520515a9R216](https://github.com/apache/nifi-minifi-cpp/pull/1879/commits/1c3a18fd8bf777b7c71a7972463e2c8b6f782f82#diff-b8f50109b2637cc95d2f47b946e4b8f961d488e508046f2922f175c5520515a9R216)